Code/GraphMol/FileParsers/MolFileParser.cpp

```diff
@@ -1005,6 +1005,34 @@ void ParseLinkNodeLine(RWMol *mol, const std::string &text, unsigned int line) {
   }
 }
 
+// Recursively populates querySet with COMPOSITE_AND queries
+// present in the input query. If the logic of the input query
+// is more complex, it returns nullptr and empty set.
+// The returned ptr should only be checked for not being null
+// and not used for any other purposes, as the actual result is
+// the querySet
+const QueryAtom::QUERYATOM_QUERY *getAndQuerySet(
+    const QueryAtom::QUERYATOM_QUERY *q,
+    std::set<const QueryAtom::QUERYATOM_QUERY *> &querySet) {
```

Comment:
the argument for using a `set` here would be in order to avoid duplicates.
Is there a chance that we'd actually end up with the same `QUERYATOM_QUERY *` being encountered more than once?

Reply:
I think it is a left-over from before fixing the "duplicate formal charge" bug. I will make it a `vector` as it can potentially hide underlying bugs. Good catch.